### PR TITLE
Remove references to legacy bro naming

### PR DESCRIPTION
**Pull request by timwoj**
This intentionally leaves behind references to `__bro_plugin__` but removes all of the other references to scripts and files that have been deprecated and/or removed since Zeek 3.0. Removing `__bro_plugin__` needs a release done with https://github.com/zeek/zeek/pull/3155 included in order to pass CI, so it will have to wait until that's available.

### CODE REVIEW

File: zkg.config [hunk 1]
```diff
@@ -66,7 +66,7 @@ bin_dir =
 
 # The directory containing Zeek distribution source code.  This is only
 # needed when installing packages that contain Zeek plugins that are
-# not pre-built.  The legacy name of this option is "bro_dist".
+# not pre-built.
```
Comment (awelzel) [inline zkg.config:69]: Today `zeek_dist` is legacy with regards to zkg and an install tree works fine for everything. That's my understanding. I'm unsure what "Zeek plugins that are note pre-built" refer to in fact.

Maybe we can note down that zeek_dist isn't needed today outside of maybe dev users?

**Review by awelzel — APPROVED** [review]
Approving, mostly left nits and comments about pre-existing things.

Address what makes sense to you, too :-)

File: zkg.config (hunk 1 above)
Comment (timwoj) [reply zkg.config:69]: Added this to the comment:

> This value is generally not needed by most users other than plugin developers anymore.

File: zeekpkg/manager.py [hunk 2]
```diff
@@ -3280,21 +3233,14 @@ def _get_package_metadata(parser):
 
 
 def _pick_metadata_file(directory):
```
[the remark this answers is not in the document]
Comment (timwoj) [reply zeekpkg/manager.py:3235]: I considered promoting it, but most callers need the value of `metadata_file` later. I got rid of the `_pick` method and just called the join directly instead.

File: zeekpkg/source.py [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
 A module containing the definition of a "package source": a git repository
-containing a collection of :file:`zkg.index` (or legacy :file:`bro-pkg.index`)
-files.  These are simple INI files that can describe many Zeek packages.  Each
-section of the file names a Zeek package along with the git URL where it is
-located and metadata tags that help classify/describe it.
+containing a collection of :file:`zkg.index` files.  These are simple INI files
+that can describe many Zeek packages.  Each section of the file names a Zeek
+package along with the git URL where it is located and metadata tags that help
+classify/describe it.
 """
```
[the remark this answers is not in the document]
Comment (timwoj) [reply zeekpkg/source.py:7]: Looks good to me.

**Pull request merged**